File: src/1437-Check-If-All-1's-Are-at-Least-Length-K-Places-Away/1437.go
func kLengthApart(nums []int, k int) bool {
    for pre, i := -k - 1, 0; i < len(nums); i++ {
        if nums[i] == 1 {
            if i - pre <= k {
                return false
            }
            pre = i
        }
    }
    return true
}

File: src/1437-Check-If-All-1's-Are-at-Least-Length-K-Places-Away/1437.cpp
class Solution {
public:
    bool kLengthApart(vector<int>& nums, int k) {
        for(int pre = -k - 1, i = 0; i < nums.size(); i++) {
            if (nums[i] == 1) {
                if (i - pre <= k) return false;
                pre = i;
            }
        }
	    return true;
    }
};